### PR TITLE
Drop python 3.7

### LINKED ISSUE #4008: Drop python 3.7?
## 🧰 Task
How does everyone feel about dropping python 3.7? 

We had previously discussed following [NEP29](https://numpy.org/neps/nep-0029-deprecation_policy.html) which:

>  recommends that all projects across the Scientific Python ecosystem adopt a common “time window-based” policy for support of Python and NumPy versions.

with a proposed support table of 

Date | Python | NumPy
-- | -- | --
Jul 26, 2021 | 3.7+ | 1.18+
Dec 22, 2021 | 3.7+ | 1.19+
Dec 26, 2021 | 3.8+ | 1.19+
Jun 21, 2022 | 3.8+ | 1.20+
Apr 14, 2023 | 3.9+ | 1.20+

with 3.8, we could get in on that sweet, sweet walrus operator, and `functools.cached_property`

### PR DESCRIPTION
**Pull request by tlambert03**
# Description
closes #4008.  no particularly hurry

## Type of change

- [ ] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

**Review by jni — APPROVED** [review]
Love it. :blush:

**Comment by jni** [comment]
Looks like the [failures](https://github.com/napari/napari/runs/5160308462?check_suite_focus=true#step:9:52) are from a missed import migration:

```pytb
  napari\components\layerlist.py:8: in <module>
      from .._vendor.cpython.functools import cached_property
  E   ModuleNotFoundError: No module named 'napari._vendor.cpython'
```

@tlambert03 can you update the PR to remove these failures?

**Comment by tlambert03** [comment]
looks like we're running into https://bugreports.qt.io/browse/PYSIDE-1140 for our pyside minreqs test with python==3.8 and pyside2==5.13.2.  After reading around a bit, the only solution i've seen is to bump the min pyside2 requirement for python 3.8 to >=5.14.0. (and i ended up using 5.14.2 to support all of the binary or combinations of Qt flags without an error).  

I also bumped pandas and scipy for `python_version >= '3.9'` to require the earliest versions that shipped wheels on pypi.

Note that pyside 5.13.2 is the latest version currently posted to conda forge.  I chatted with @jaimergp about this and while this is unrelated to this PR, just want to leave a note here:  Essentially, we can't use PySide2 for our conda-based bundle until one of https://github.com/conda-forge/pyside2-feedstock/pulls merges.  As I understand it, the conda bundles have been working so far because on of the dependencies (qtconsole) had been pulling in pyqt, and it was all using pyqt under the hood.  (Technically, we could use pyside2 5.13.2 in the bundle if we used python 3.7, but nobody wants that...)

Anyway, as mentioned, that's more or less unrelated to whether we drop 3.7,  and assuming minreqs now passes, this should be ready to go.

**Review by sofroniewn — APPROVED** [review]
LGTM! Goodbye 3.7 and 600+ lines of code 🚀  cc @napari/core-devs. Thanks!!

File: napari/__main__.py [hunk 1]
```diff
@@ -25,8 +25,7 @@ def __call__(self, *args, **kwargs):
         from .plugins import plugin_manager
 
         plugin_manager.discover_widgets()
-        errors = plugin_manager.get_errors()
```
Comment (brisvag) [inline napari/__main__.py:28]: <3 :=

**Review by brisvag — APPROVED** [review]
LGTM!

**Pull request merged**